File: webshop-spring/src/main/java/org/nicolai/webshop/service/UserService.java
package org.nicolai.webshop.service;

import org.nicolai.webshop.model.User;
import org.nicolai.webshop.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {

    @Autowired
    UserRepository userRepository;

    @Autowired
    PasswordEncoder passwordEncoder;

    public User registerUser (User user) {
        //System.out.println(user);
        try {
            user.setPassword(passwordEncoder.encode(user.getPassword()));
            User newUser =  userRepository.save(user);
            //System.out.println(newUser);
            return newUser;
        } catch (Exception e) {
            System.out.println(e.getMessage());
            return null;
        }


    }

    public List<User> getAll() {
        return userRepository.findAll();
    }

    public User getById(int id) {
        var user = userRepository.findById(id);
        return user.orElse(null);

    }

    public User update(int id, User user) {
        user.setId(id);
        user.setPassword(passwordEncoder.encode(user.getPassword()));
        return userRepository.save(user);
    }

    public User delete(int id) {
        var user = userRepository.findById(id);
        userRepository.deleteById(id);
        return user.orElse(null);
    }
}

File: bank-spring/src/main/kotlin/org/nicolai/bank/service/UserService.kt
package org.nicolai.bank.service

import org.nicolai.bank.dto.UserDto
import org.nicolai.bank.repository.UserRepository
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.stereotype.Service

@Service
class UserService(val userRepository: UserRepository) {

    fun getOne(id: Int): UserDto? {
        return userRepository.getOne(id)
    }

    fun getAll(): List<UserDto> {
        return userRepository.getAll()
    }

    fun addUser (user: UserDto): UserDto? {

        return userRepository.addUser(user)

    }

    fun updateOne(id: Int, user: UserDto): UserDto {
        return userRepository.updateOne(id, user)
    }

    fun deleteOne(id: Int): UserDto? {
        return userRepository.deleteOne(id)
    }


}
